cmd/influx/v1_authorization.go

```diff
@@ -31,155 +31,175 @@ func cmdV1Auth(f *globalFlags, opt genericCLIOpts) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "create",
 		Short: "Create authorization",
-		RunE:  checkSetupRunEMiddleware(&flags)(v1AuthorizationCreateF),
+		RunE:  checkSetupRunEMiddleware(&flags)(makeV1AuthorizationCreateE(opt)),
 	}
 
 	f.registerFlags(cmd)
 	v1AuthCreateFlags.org.register(cmd, false)
 
-	cmd.Flags().StringVar(&v1AuthCreateFlags.token, "auth-token", "", "Token, formatted as username:password")
-	cmd.MarkFlagRequired("auth-token")
+	cmd.Flags().StringVar(&v1AuthCreateFlags.username, "username", "", "The username to identify this token")
+	_ = cmd.MarkFlagRequired("username")
 	cmd.Flags().StringVarP(&v1AuthCreateFlags.description, "description", "d", "", "Token description")
-	cmd.Flags().StringVarP(&v1AuthCreateFlags.user, "user", "u", "", "The user name")
-	registerPrintOptions(cmd, &v1AuthCRUDFlags.hideHeaders, &v1AuthCRUDFlags.json)
+	cmd.Flags().BoolVar(&v1AuthCreateFlags.noPassword, "no-password", false, "Don't set password. You must use v1 auth set-password command before using token.")
+	registerPrintOptions(cmd, &v1AuthCreateFlags.hideHeaders, &v1AuthCreateFlags.json)
 
 	cmd.Flags().StringArrayVarP(&v1AuthCreateFlags.writeBucketPermissions, "write-bucket", "", []string{}, "The bucket id")
 	cmd.Flags().StringArrayVarP(&v1AuthCreateFlags.readBucketPermissions, "read-bucket", "", []string{}, "The bucket id")
 
 	return cmd
 }
 
-func v1AuthorizationCreateF(cmd *cobra.Command, args []string) error {
-	if err := v1AuthCreateFlags.org.validOrgFlags(&flags); err != nil {
-		return err
-	}
+func makeV1AuthorizationCreateE(opt genericCLIOpts) func(*cobra.Command, []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		if err := v1AuthCreateFlags.org.validOrgFlags(&flags); err != nil {
+			return err
+		}
 
-	userSvc, err := newUserService()
-	if err != nil {
-		return err
-	}
+		userSvc, err := newUserService()
+		if err != nil {
+			return err
+		}
 
-	orgSvc, err := newOrganizationService()
-	if err != nil {
-		return err
-	}
+		orgSvc, err := newOrganizationService()
+		if err != nil {
+			return err
+		}
 
-	orgID, err := v1AuthCreateFlags.org.getID(orgSvc)
-	if err != nil {
-		return err
-	}
+		orgID, err := v1AuthCreateFlags.org.getID(orgSvc)
+		if err != nil {
+			return err
+		}
 
-	bucketPerms := []struct {
-		action influxdb.Action
-		perms  []string
-	}{
-		{action: influxdb.ReadAction, perms: v1AuthCreateFlags.readBucketPermissions},
-		{action: influxdb.WriteAction, perms: v1AuthCreateFlags.writeBucketPermissions},
-	}
+		s, err := newV1AuthorizationService()
+		if err != nil {
+			return err
+		}
 
-	var permissions []influxdb.Permission
-	for _, bp := range bucketPerms {
-		for _, p := range bp.perms {
-			var id influxdb.ID
-			if err := id.DecodeFromString(p); err != nil {
+		// verify an existing token with the same username doesn't already exist
+		filter := influxdb.AuthorizationFilter{
+			Token: &v1AuthCreateFlags.username,
```

Comment:
I find it confusing that the `--username` flag gets mapped to the token name, instead of `user.Name`. Ex: in the demo video the command `v1 auth create --username bar` creates a token with: `Name / Token: bar` and `User Name: my-user`. It's not clear where that `my-user` came from, or how I could set it if I wanted to.

Reply:
@danxmoran I agree. However, this is an artefact of existing product deficiencies, product management requirements (upgrade 1.x without touching existing clients) and maintaining existing API compatibility with Cloud by not changing existing Users or Tokens.

1. Product management was forced to use tokens as the User object does not provide the appropriate features to restrict access to a subset of buckets, whereas Tokens do.

2. Tokens are always owned / associated with a real v2 `User` object. This is the `User Name` column and the `my-user` user you identified. By default, when you access an InfluxDB instance via the `influx` CLI tool, you are using a token owned by a v2 User (set via config or an environment variable), in this case `my-user`. So, when the operator create new tokens, they will be owned by their V2 User.

Reply:
Hmmm in that case, could we use something like `--token-name` as the option instead of `--username`? Or does `username` have a specific meaning when it comes to V1 that we shouldn't change?

Reply:
No, `--token-name` is unclear. `--username` is familiar to a v1 user, as this will match their users upgraded from a v1 database.

Reply:
Fair enough 😄

Reply:
To be clear, users upgraded from v1 are _not_ created as real V2 users, due to the deficiencies I described earlier. They will be created as these special tokens.

Reply:
Our hands a tied somewhat, as we can't leverage the existing V2 Tokens. We chose to create these private tokens, with restricted use only to the 1.x compatible APIs (`/write` and `/query`).

1. We can't change them and break compatibility with Cloud APIs (for example to support bcrypt passwords, for a seamless upgrade from 1.x to 2.x)
2. It is time-consuming to propose a new token type, requiring a thorough review and verification they will be viable and useful in Cloud. Time constraints for OSS 2.0 GA prevented us from taking this approach.